### DISCUSSION TITLE
Full Translation

### DISCUSSION THREAD

**Discussion by Odilbukh** [filamentphp/filament#2747] [General]
Filament is very usefull and good tool!

But about translation in documention very les info. There is no examples.

For example I could not find info, about how to translate model, recourse name, fields and so on from documentation.

So what to do?

**Comment by danharrin** [comment]
https://filamentphp.com/docs/2.x/admin/resources/getting-started#labels
https://filamentphp.com/docs/2.x/forms/fields#setting-a-label
https://filamentphp.com/docs/2.x/tables/columns#setting-a-label

**Reply by Odilbukh** [reply]
Thank you! It is usefull info
Now I can translate label and so on!

I have translated label and used getLabel() in resource. It works good. But it adds 's' to label name in sidebar (not only sidebar). 
Pls see the screenshot. Can you help me to fix it?

![image](https://user-images.githubusercontent.com/22895615/173358927-88f780bc-bb7d-4326-b5d3-6cbe188ce47e.png)

**Reply by danharrin** [reply]
Override the `getPluralLabel()`, it's in the docs

**Reply by Odilbukh** [reply]
Thanks!

**Reply by alive2tinker** [reply]
@danharrin  where does filament look for translation? I have ar.json file and it definitely doesn't  look there. where should I store the file and in which format?

my main issue is when I use `Tables\Columns\TextColumn::make('column_name')->label('Column Name')->translateLabel()` and it doesn't translate the column name

**Reply by danharrin** [reply]
You can always use `__()` instead of `translateLabel()`

**Reply by alive2tinker** [reply]
I have tried that and it doesn't seem to catch my translations. I store all my translations in `lang/ar.json` and it never worked. I saw this [Thread](https://github.com/filamentphp/filament/discussions/2103) where mentioned it is not a good practice, and I'm wondering what is the right practice for filament?

**Reply by danharrin** [reply]
Well firstly I would advise using `.php` files instead of `.json`, with a nested folder structure that mimics your application. Then you translate generic strings instead of their direct English versions... like

`__('resource/table.columns.column_name.label')`

**Reply by alive2tinker** [reply]
I took your advice and now using .php file, but translation is still not working. here is what I did:
1- I create PostResource using `php artisan make:filament-resource PostResource --generate --soft-deletes`
2- I created the file `table.php` and put inside the resource
/app
|-- Filament
|.         |--   Resources
|          |          |-- PostResource 
|          |          |               |-- table.php
3- I created the column using this line
`Tables\Columns\TextColumn::make('title')->label(__('PostResource/table.columns.title.label'))->translateLabel()`    
4- the result comes to  this `PostResource/table.columns.title.label`


@danharrin please let me know what i'm missing here.

**Reply by alive2tinker** [reply]
I have also just tried the following:
1- put `table.php` directly inside `lang` folder. does not work
2- put `table.php` directly inside `lang/ar/` folder. does not work
3- put `table.php` directly inside filament translation folder. does not work

**Reply by alive2tinker** [reply]
figured it out. I believe filament is looking for translations in `resources/lang` while laravel has moved the lang folder outside resources.

**Reply by danharrin** [reply]
hmm no, both should work
